public/app/features/alerting/unified/components/settings/AlertmanagerConfig.tsx

```diff
@@ -0,0 +1,185 @@
+      try {
+        JSON.parse(value);
+        return true;
+      } catch (e) {
+        return e instanceof Error ? e.message : 'JSON is invalid';
+      }
+    },
+  });
+
+  const handleSave = handleSubmit(
+    (values: FormValues) => {
+      onSave(alertmanagerName, defaultValues.configJSON, values.configJSON);
+    },
+    (errors) => {
+      console.error(errors);
+    }
+  );
+
+  const isOperating = isLoadingConfig || isDeleting || isSaving;
+
+  /* loading error, if this fails don't bother rendering the form */
+  if (loadingError) {
+    return (
+      <Alert severity="error" title="Failed to load Alertmanager configuration">
+        {loadingError.message ?? 'An unkown error occurred.'}
+      </Alert>
+    );
+  }
+
+  /* resetting configuration state */
+  if (isDeleting) {
+    return (
+      <Alert severity="info" title="Resetting Alertmanager configuration">
+        It might take a while...
+      </Alert>
+    );
+  }
+
+  return (
+    <div className={styles.container}>
+      {/* form error state */}
+      {errors.configJSON && (
+        <Alert severity="error" title="Oops, something went wrong">
+          {errors.configJSON.message || 'An unknown error occurred.'}
+        </Alert>
+      )}
+
+      {isLoadingSuccessful && (
+        <div className={styles.content}>
+          <AutoSizer>
+            {({ height, width }) => (
+              <>
+                <CodeEditor
+                  language="json"
+                  width={width}
+                  height={height}
+                  showLineNumbers={true}
+                  monacoOptions={{
+                    scrollBeyondLastLine: false,
+                  }}
+                  value={defaultValues.configJSON}
+                  showMiniMap={false}
+                  onSave={(value) => setValue('configJSON', value)}
+                  onBlur={(value) => setValue('configJSON', value)}
+                  readOnly={isOperating}
+                />
+              </>
+            )}
+          </AutoSizer>
+        </div>
+      )}
+
+      {!readOnly && (
+        <Stack justifyContent="flex-end">
+          <Button variant="destructive" disabled={isOperating} onClick={() => setShowResetConfirmation(true)}>
+            Reset
+          </Button>
+          <Spacer />
+          <Button variant="secondary" onClick={() => onDismiss()} disabled={isOperating}>
+            Cancel
+          </Button>
+          <Button variant="primary" onClick={handleSave} disabled={isOperating}>
+            Save
+          </Button>
+        </Stack>
+      )}
+      <ConfirmModal
+        isOpen={showResetConfirmation}
+        title="Reset Alertmanager configuration"
+        body={`Are you sure you want to reset configuration ${
+          alertmanagerName === GRAFANA_RULES_SOURCE_NAME ? 'for the Grafana Alertmanager' : `for "${alertmanagerName}"`
+        }? Contact points and notification policies will be reset to their defaults.`}
```

Comment:
nit: I believe constructing the text in this way makes it harder to translate later on. I _think_ its better in this case to have two strings something like this:

```
  const foo = `Are you sure you want to reset configuration for the Grafana Alertmanager? Contact points and notification policies will be reset to their defaults.`;

  const bar = `Are you sure you want to reset configuration for "${alertmanagerName}"? Contact points and notification policies will be reset to their defaults.`;
```